#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Allow ldconfig to work on i386/x86-64 biarch systems
# DP: Related bugs: 
# DP: Dpatch author: Daniel Jacobowitz
# DP: Patch author: Daniel Jacobowitz
# DP: Upstream status: Debian-Specific
# DP: Status Details: based on Ubuntu change by Jeff Bailey
# DP: Date: 2005-10-13

PATCHLEVEL=1

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- /dev/null	2005-07-29 18:11:21.000000000 +0000
+++ glibc-2.3.5/sysdeps/unix/sysv/linux/i386/dl-cache.h	2005-08-05 14:15:39.162794976 +0000
@@ -0,0 +1,1 @@
+#include <sysdeps/unix/sysv/linux/sparc/dl-cache.h>
